crates/esthri/tests/integration/sync_test.rs

```diff
@@ -189,6 +189,202 @@ fn test_sync_up_default() {
     }
 }
 
+// Validate local to remote sync with delete flag set
+#[test]
+fn test_sync_up_delete() {
+    let s3client = esthri_test::get_s3client();
+    let local_directory = esthri_test::copy_test_data("sync_up");
+    let s3_key_prefix = esthri_test::randomised_name("test_sync_up_delete/");
+
+    let source = S3PathParam::new_local(&local_directory);
+    let destination = S3PathParam::new_bucket(esthri_test::TEST_BUCKET, &s3_key_prefix);
+
+    let res = blocking::sync(
+        s3client.as_ref(),
+        source.clone(),
+        destination.clone(),
+        FILTER_EMPTY,
+        false,
+        true,
+    );
+    assert!(res.is_ok());
+
+    let keys = ["1-one.data", "2-two.bin", "3-three.junk", "nested/2MiB.bin"];
+
+    for key in &keys[..] {
+        let key = format!("{}{}", &s3_key_prefix, key);
+        let res = blocking::head_object(s3client.as_ref(), esthri_test::TEST_BUCKET, &key);
+        assert!(res.is_ok(), "head object failed for: {}", key);
+        let res = res.unwrap();
+        assert!(
+            res.is_some(),
+            "head object info returned was nil for: {}",
+            key
+        );
+    }
+
+    let remove_path_target = "nested/2MiB.bin";
+    let remove_path_target = local_directory.join(remove_path_target);
+
+    fs::remove_file(&remove_path_target).expect("could not remove file");
+    assert!(fs::metadata(remove_path_target).is_err());
```

Comment:
I don't think you need to assert that the file is gone right after you call remove_file. Also consider using a temporary file instead of manually performing clean up

Reply:
I agree with the unnecessary assert you described above. 
This file needs to have a specific name, which tempfile does not provide. a file could be joined to a tempDIr, but that would require refactoring that would affect the creation of random file prefixes and I do not know what the implications of this are to esthri.